practices/understanding_git.md

```diff
@@ -0,0 +1,105 @@
+
+
+
+## Why Git Is Preferred
+
+- Git is much faster than Subversion.
+- Git's repositories are much smaller than Subversions.
+- Git was designed to be fully distributed from the start, allowing each developer to have full local control.
+- Git branches are simpler and less resource heavy than Subversion's.
+- Git branches carry their entire history.
+- Git provides better auditing of branch and merge events
+- Git's repo file formats are simple, so repair is easy and corruption is rare.
+- Git repository clones act as full repository backups.
+
+## Basics of Using Git
+There are a lot of ways that you can start. For your guidance, I have compiled which I call a noob guide to start with git.🙂
+
+You can start by following these steps:
+
+1. First, download git. For windows, you can download it from [here](https://git-scm.com/downloads). For Linux use this command `sudo apt-get update $ sudo apt-get install git`. For MAC you can download it from [here](https://sourceforge.net/projects/git-osx-installer/files/).
+2. Create an empty repository by using the `git init` command or you can checkout a repository by `git clone /path/to/repository` command while using a remote server you can use `git clone username@host:/path/to/repository`.
+3. Now make the changes in the repository according to your need. (This step will follow the workflow that was discussed above)
+4. The changes made by you can be proposed (add to **INDEX**) using `git add <filename>`.
+5. To commit these changes you can use `git commit -m "Commit message"`.
+6. Now the changes that you have made are in the **head** to send it to the remote repository use `git push origin master`. If you have not cloned the repository you can use `git remote add origin <server>`.
+7. If you want to add a feature to your project but don't want to disturb the original project you can make a branch out of the main project by using `git checkout -b feature_x` command. You can switch to the original project by using the `git checkout master` command. Remember unless you push the branch using `git push origin <branch>` it will not be publicly available.
+
+
+Hurrayy!! you created your first branch commit with knowing what exactly goes in the backend.🎉
+
+## Some Pitfalls of the Git 
+
+- User interface maturity
```

Comment:
What do you mean by this User interface maturity?

Reply:
Currently Subversion has a wider range of user interface tools than Git. For example there are SVN plugins available for most popular IDEs. There is a Windows Explorer shell extension. There are a number of native Windows and Mac OS X GUI tools available in ready-to-install packages.

Git's primary user interface is through the command line. There are two graphical interfaces: git-gui (distributed with Git) and qgit, which is making great strides towards providing another feature-complete graphical interface. Also gitk, the graphical history browser, can be more than just a fancy log reader. git-gui and gitk usually work out-of-box for common operating systems, and qgit is being ported to Qt4, which improves its portability. There are some user interface tools in development for Git, namely TortoiseGit, a port of TortoiseSVN. There is also Git Extensions, another explorer shell extension

Reply:
I see now, but the User interface maturity is not a clear term by itself. Also, did you consider the graphical tools I have mentioned in a different comment (Fork, Github Desktop, GitKraken, and others)? 

I would suggest something like "A developer doesn't know how to use the command-line git interface."